src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService_IncrementalAnalyzer.cs

```diff
@@ -218,7 +197,8 @@ internal BaseDiagnosticIncrementalAnalyzer Analyzer
 
             private BaseDiagnosticIncrementalAnalyzer GetAnalyzer(bool useV2)
             {
-                return useV2 ? (BaseDiagnosticIncrementalAnalyzer)_engineV2 : _engineV1;
+                // v1 engine has been removed, always use v2 engine.
```

Comment:
Hmm, why should this throw if we are changing the option from Tools Option or registry key? Shouldn't it just be a no-op?